src/coreclr/jit/lower.cpp

```diff
@@ -5190,10 +5190,34 @@ bool Lowering::LowerUnsignedDivOrMod(GenTreeOp* divMod)
         int    postShift;
         bool   simpleMul = false;
 
+        unsigned bits = type == TYP_INT ? 32 : 64;
+        // if the dividend operand is AND or RSZ with a constant then the number of input bits can be reduced
+        if (dividend->OperIs(GT_AND))
+        {
+            size_t maskCns = static_cast<size_t>(
+                dividend->gtGetOp1()->IsCnsIntOrI()
+                    ? dividend->gtGetOp1()->AsIntCon()->IconValue()
+                    : dividend->gtGetOp2()->IsCnsIntOrI() ? dividend->gtGetOp2()->AsIntCon()->IconValue() : 0);
```

Comment:
I do not think it is worth checking for constant `op1` here, `gtSetEvalOrder` always commutes them to the right.

(Getting rid of it will also get of the nested ternary, which the code guide says is not allowed, and merge the two `if`s, potentially).